### DISCUSSION TITLE
Having trouble with get() when getting a putSigned() value.

### DISCUSSION THREAD

**Discussion by n7tae** [savoirfairelinux/opendht#645] [Q&A] [answered]
I think I'm doing something wrong. On my new OpenDHT network, get() is returning multiple values that were published using putSigned(). I though if a node uses putSigned(), that node owns that key-value and there should only be one value, even if it's updated by the node. get() is returning multiple values with different dht::Value::id. How can there be different values on my dht?

**Comment by aberaud** [comment, marked as answer]
Using putSigned would not be enough to make the value unique and allow to edit it.
The ValueID also has to be the same.

To do that:
* Set a random non-0 `id` for the `Value` before the first `putSigned`, or let OpenDHT assign a random ID to the `Value`, then modify/reuse the same Value for the following `putSigned`
* or build different Value objects but manually set the same non-0 value ID on them

If the value ID is 0 when putSigned is called, the value will be assigned a random ID during the putSigned (set by OpenDHT on the Value at some point between putSigned() and the done callback).
